server/api/src/rsDataObjClose.cpp

```diff
@@ -449,16 +450,42 @@ namespace
         auto r = ir::make_replica_proxy(*l1desc.dataObjInfo);
 
         try {
-            const auto size_should_be_verified    = !getValByKey(&l1desc.dataObjInp->condInput, NO_CHK_COPY_LEN_KW);
-            const auto replica_opened_for_write   = OPEN_FOR_WRITE_TYPE == l1desc.openType || CREATE_TYPE == l1desc.openType;
-            const auto replica_has_a_checksum     = !r.checksum().empty();
-            const auto checksum_should_be_updated = replica_opened_for_write && replica_has_a_checksum;
+            if (PUT_OPR == l1desc.oprType) {
+                if (REG_CHKSUM == l1desc.chksumFlag || VERIFY_CHKSUM == l1desc.chksumFlag) {
+                    // Update the replica proxy's checksum value so that if an exception is thrown,
+                    // the checksum value will still be stored in the catalog.
+                    r.checksum(irods::register_new_checksum(_comm, *r.get(), l1desc.chksum));
+                    rst::update(r.data_id(), r);
+
+                    if (VERIFY_CHKSUM == l1desc.chksumFlag) {
+                        if (r.checksum() != l1desc.chksum) {
+                            THROW(USER_CHKSUM_MISMATCH, fmt::format("[{}:{}] mismatch checksum for {}.inp={}, compute {}",
+                                  __FUNCTION__, __LINE__, r.logical_path(), l1desc.chksum, r.checksum()));
+                        }
+                    }
+
+                    // Copy the new checksum value into the conditional input so that file modified is triggered.
+                    cond_input[CHKSUM_KW] = r.checksum();
+                }
 
-            if (checksum_should_be_updated) {
-                l1desc.chksumFlag = REG_CHKSUM;
+                return;
             }
 
-            if (checksum_should_be_updated || size_should_be_verified) {
+            // clang-format off
+            const auto size_should_be_verified    = !getValByKey(&l1desc.dataObjInp->condInput, NO_CHK_COPY_LEN_KW);
+            //const auto replica_opened_for_write   = OPEN_FOR_WRITE_TYPE == l1desc.openType || CREATE_TYPE == l1desc.openType;
+            //const auto replica_has_a_checksum     = !r.checksum().empty();
+            //const auto checksum_should_be_updated = replica_opened_for_write && replica_has_a_checksum;
+            // clang-format on
+
+            // TODO This appears to be the auto-update case. Lets disable it and see what CI
+            // complains about.
+            //if (checksum_should_be_updated) {
+                //l1desc.chksumFlag = REG_CHKSUM;
+            //}
+
+            //if (checksum_should_be_updated || size_should_be_verified) {
```

Comment:
This code is safe to remove given that this PR passed CI.

Reply:
agreed

Reply:
done